src/client/Microsoft.Identity.Client/ManagedIdentityApplication.cs

```diff
@@ -54,5 +63,39 @@ public AcquireTokenForManagedIdentityParameterBuilder AcquireTokenForManagedIden
                 ClientExecutorFactory.CreateManagedIdentityExecutor(this),
                 resource);
         }
+
+        /// <summary>
+        /// Used to determine if managed identity is able to perform Proof-of-Possession.
+        /// </summary>
+        /// <returns>Boolean indicating if Proof-of-Possession is supported</returns>
+        public bool IsProofOfPossessionSupportedByClient()
+        {
+            return KeyMaterialManager.CryptoKeyType == CryptoKeyType.KeyGuard;
+        }
+
+        /// <summary>
+        /// Used to determine if managed identity is able to handle claims.
+        /// </summary>
+        /// <returns>Boolean indicating if Claims is supported</returns>
+        public bool IsClaimsSupportedByClient()
+        {
+            return KeyMaterialManager.CryptoKeyType != CryptoKeyType.None;
+        }
+
+        /// <summary>
+        /// Retrives the binding certificate for advanced managed identity scenarios.
+        /// </summary>
+        /// <returns>Binding certificate used for advanced scenarios</returns>
+        public static X509Certificate2 GetBindingCertificate()
```

Comment:
Yes multiple uses, For CAE this will be a helper for Azure Identity to get the cert if they want to customize httpclient, for POP this will be the helper to get the binding cert for resource communication. We also plan to expose this in AuthResult when we GA